src/main/scala/com/exasol/spark/util/ExasolConnectionManager.scala

```diff
@@ -23,18 +23,26 @@ import com.exasol.jdbc.EXAStatement
  */
 final case class ExasolConnectionManager(config: ExasolConfiguration) {
 
-  private[this] val mainJdbcConnectionUrl = s"jdbc:exa:${config.host}:${config.port}"
+  private[this] val DO_NOT_VALIDATE_CERTIFICATE = "validateservercertificate=0"
+  private[this] val MAIN_CONNECTION_PREFIX = "jdbc:exa"
+  private[this] val WORKER_CONNECTION_PREFIX = "jdbc:exa-worker"
 
   /** A regular Exasol jdbc connection string */
-  def getJdbcConnectionString(): String =
-    getConnectionStringWithOptions(mainJdbcConnectionUrl)
+  def getJdbcConnectionString(): String = {
+    val host = getHostWithFingerprint(config.host)
+    val url = s"$MAIN_CONNECTION_PREFIX:$host:${config.port}"
+    getConnectionStringWithOptions(url)
+  }
+
+  private[this] def getHostWithFingerprint(host: String): String =
+    if (!config.fingerprint.isEmpty() && !config.jdbc_options.contains(DO_NOT_VALIDATE_CERTIFICATE)) {
+      host + "/" + config.fingerprint
```

Comment:
using Paths.get() to obtain the desired path would be better

Reply:
Here it is not actually a path, but change to the host address that is provided by user, e.g, 10.10.0.1.